addons/sale/static/src/js/sale_order_line_mixin.js

```diff
@@ -0,0 +1,18 @@
+odoo.define('sale.UpdateAllLinesMixin', function (require) {
+    'use strict';
+
+    const OpenUpdateAllWizardMixin = {
+        _onFieldChanged: function (ev) {
+            if (ev && ev.data.changes) {
+                const fieldName = Object.keys(ev.data.changes)[0];
+                const value = ev.data.changes[fieldName];
+                this.trigger_up(this._getUpdateAllLinesAction(), {fieldName: fieldName, value: value});
+            }
+            this._super.apply(this, arguments);
+        },
```

Comment:
Yes, for now it is only used/working on sale order lines. If there is demands, it could be adapted to be a core feature. I don't think it's necessary right now. 

Indeed, it only works on the displayed line (first 40).  The advantage to use it in JS is to allow discard changes easily. It can also work on new lines not existing in DB. I don't know if this 40 limit is a real issue if this functionality remains confined to sale order lines. If it was generalized, it would be an issue for sure.